docs/data-companion/pruning.md

```diff
@@ -0,0 +1,338 @@
+# Configuration specifically for the gRPC pruning service, which is considered a
+# privileged service.
+#
+[grpc.privileged.pruning_service]
+
+# Only controls whether the pruning service is accessible via the gRPC API - not
+# whether a previously set pruning service retain height is honored by the
+# node. See the [storage.pruning] section for control over pruning.
+#
+# Disabled by default.
+enabled = true
+```
+
+## Pruning configuration
+
+Ensure that the data companion pruning is enabled in the configuration to allow the data companion to influence the
+node pruning mechanism.
+
+In the `[storage.pruning.data_companion]` section of the CometBFT's configuration
+file, the property `enabled` should be set to `true`:
+
+```
+[storage.pruning.data_companion]
+
+# Whether automatic pruning respects values set by the data companion. Disabled
+# by default. All other parameters in this section are ignored when this is
+# disabled.
+#
+# If disabled, only the application retain height will influence block pruning
+# (but not block results pruning). Only enabling this at a later stage will
+# potentially mean that blocks below the application-set retain height at the
+# time will not be available to the data companion.
+enabled = true
+```
+
+In order to avoid unwanted pruning of data when the data companion is activated, it is possible to define the initial
+retain height for block and block results in the configuration file. This configuration ensures that the necessary data
+is retained and not removed in an undesirable way.
+
+For example, you can change the values in this part of the configuration:
+```
+# The initial value for the data companion block retain height if the data
+# companion has not yet explicitly set one. If the data companion has already
+# set a block retain height, this is ignored.
+initial_block_retain_height = 0
+
+# The initial value for the data companion block results retain height if the
+# data companion has not yet explicitly set one. If the data companion has
+# already set a block results retain height, this is ignored.
+initial_block_results_retain_height = 10
+```
+
+## Retain Height
+
+One important concept that can affect the pruning of nodes is the `retain height`. The retain height determines the specific
+height from which the data can be safely deleted from the node's storage. By considering the retain height,
+nodes can effectively manage their storage usage and ensure that they are only retaining the data that is necessary for
+their operations. This is important because storage space is a finite resource and nodes with limited storage space may
+struggle to keep up with the growth of the blockchain.
+
+## Pruning Blocks
+
+The pruning service uses the "block retain height" parameter to specify the height to which the node will
+preserve blocks. It is important to note that this parameter differs from the application block retain height, which
+the application sets in response to ABCI commit messages.
```

Comment:
I would add another sentence here of the form :
When deciding which blocks to prune, CometBFT choses the lower of the two retain heights. By default the application retain height is 0, which means that no blocks will be pruned until the application changes this. The default retain height for an enabled data companion is 0 as well. This ensures that the application will not trigger pruning until the data companion has done the same.